### PR TITLE
Tests find median run

### PR DESCRIPTION
**Pull request by pedro93**
As part of https://github.com/paulirish/pwmetrics/issues/55 addresses https://github.com/paulirish/pwmetrics/issues/72

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -200,13 +199,29 @@ class PWMetrics {
...
+
+    values.sort(function sortValues(a, b) {
+      return a - b;
+    });
+
+    const half = Math.floor(values.length / 2);
+    if (values.length % 2)
+      return values[half];
+    else
+      return values[half]; // take slightly worse entry as median.
```
Comment (staabm) [inline lib/index.js]: IF and ELSE branch contain the same return value?

Comment (pedro93) [inline lib/index.js]: It's a mistake, thanks for catching it, just commited a fix. There was also the case where values.length is 2 and the half variable already is correct index.

File: lib/index.js [hunk 2]
```diff
@@ -200,13 +199,29 @@ class PWMetrics {
...
     }));
     return medianRun;
   }
+
+  getMedianRun(values) {
+    if (values.length == 1) {
+      return values[0];
+    }
+
+    values.sort(function sortValues(a, b) {
```
Comment (denar90) [inline lib/index.js]: just arrow func
`values.sort((a, b) => a-b);`

File: lib/index.js [hunk 3]
```diff
@@ -200,13 +199,29 @@ class PWMetrics {
...
+      return a - b;
+    });
+
+    const half = Math.floor(values.length / 2);
+    if (values.length % 2 || values.length === 2)
+      return values[half];
+    else
+      return values[half+1]; // take slightly worse entry as median.
```
Comment (denar90) [inline lib/index.js]: `half + 1`

File: lib/index.js [hunk 4]
```diff
@@ -200,13 +199,29 @@ class PWMetrics {
...
+    const medianTTI = this.getMedianRun(ttiValues);
     // in the case of duplicate runs having the exact same TTI, we naively pick the first
     const medianRun = results.find(result => result.timings.find(timing => {
       return timing.id === metrics.ids.TTI && timing.timing === medianTTI;
     }));
     return medianRun;
   }
+
+  getMedianRun(values) {
+    if (values.length == 1) {
```
Comment (denar90) [inline lib/index.js]: `===`

File: package.json [hunk 5]
```diff
@@ -22,7 +22,6 @@
     "google-auth-library": "^0.10.0",
     "googleapis": "^16.0.0",
     "lighthouse": "1.6.0",
-    "median": "^0.0.2",
```
Comment (paulirish) [inline package.json]: we also have a yarn lock that'll need an update with this.

File: lib/index.js [hunk 6]
```diff
@@ -200,13 +199,25 @@ class PWMetrics {
 
   findMedianRun(results) {
     const ttiValues = results.map(r => r.timings.find(timing => timing.id === metrics.ids.TTI).timing);
-    const medianTTI = median(ttiValues);
+    const medianTTI = this.getMedianRun(ttiValues);
     // in the case of duplicate runs having the exact same TTI, we naively pick the first
     const medianRun = results.find(result => result.timings.find(timing => {
```
Comment (paulirish) [inline lib/index.js]: a little odd we now have `findMedianRun()` `getMedianRun()` and `medianRun`.

afaict, your new `getMedianRun` is just getting a median value of the TTIs passed in, but not returning a run.

can we clean this up?

i'm open to solutions, but one would be we nuke this last `results.find(result => result... ` bit and and pass the `results` into your new getMedianRun. In there, we operate on each `result` and compare its TTI result value to the other. and in the end we return the result object.

but if you want do it differently, that WFM too. :)

File: package.json (hunk 5 above)
Comment (pedro93) [inline package.json]: Just committed a my proposal for the code change, when you have a minute let me know if you think it's ok @paulirish

File: lib/index.js (hunk 6 above)
Comment (pedro93) [inline lib/index.js]: Just committed my proposal for the code change, when you have a minute let me know if you think it's ok @paulirish

File: package.json (hunk 5 above)
Comment (pedro93) [inline package.json]: done

File: lib/index.js (hunk 3 above)
Comment (pedro93) [inline lib/index.js]: done

File: lib/index.js (hunk 4 above)
Comment (pedro93) [inline lib/index.js]: done

File: lib/index.js (hunk 2 above)
Comment (pedro93) [inline lib/index.js]: done

File: lib/index.js (hunk 1 above)
Comment (pedro93) [inline lib/index.js]: done

File: lib/index.js (hunk 6 above)
Comment (pedro93) [inline lib/index.js]: done

**Pull request merged**